### PR TITLE
fix: Define "scheduler_address" in SimpleScalable mode

### LINKED ISSUE #12506: Loki 3.0 Feedback and Issues
If you encounter any troubles upgrading to Loki 3.0 or have feedback for the upgrade process, please leave a comment on this issue!

Also you can ask questions at: https://slack.grafana.com/ in the channel `#loki-3`

### PR DESCRIPTION
**Pull request by maksym-iv**
**What this PR does / why we need it**:

It looks like "SimpleScalable" deployment mode still needs non-empty `schedulerAddress` helper value to be used in the `frontend.scheduler_address` and `frontend_worker.scheduler_address` config, currently it is not empty string only if mode is Distributed

Note, `query-scheduler-discovery` service name is hardcoded in the `production/helm/loki/templates/backend/query-scheduler-discovery.yaml`

**Which issue(s) this PR fixes**:
Related to #12506
Comment https://github.com/grafana/loki/issues/12506#issuecomment-2050508770

**Special notes for your reviewer**:

**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [x] Documentation added
- [x] Tests updated
- [x] Title matches the required conventional commits format, see [here](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] Changes that require user attention or interaction to upgrade are documented in `docs/sources/setup/upgrade/_index.md`
- [x] For Helm chart changes bump the Helm chart version in `production/helm/loki/Chart.yaml` and update `production/helm/loki/CHANGELOG.md` and `production/helm/loki/README.md`. [Example PR](https://github.com/grafana/loki/commit/d10549e3ece02120974929894ee333d07755d213)
- [x] If the change is deprecating or removing a configuration option, update the `deprecated-config.yaml` and `deleted-config.yaml` files respectively in the `tools/deprecated-config-checker` directory. [Example PR](https://github.com/grafana/loki/pull/10840/commits/0d4416a4b03739583349934b96f272fb4f685d15)

### CODE REVIEW

File: production/helm/loki/templates/_helpers.tpl [hunk 1]
```diff
@@ -943,8 +943,11 @@ enableServiceLinks: false
 {{- define "loki.querySchedulerAddress" -}}
 {{- $schedulerAddress := ""}}
 {{- $isDistributed := eq (include "loki.deployment.isDistributed" .) "true" -}}
+{{- $isSimpleScalable := eq (include "loki.deployment.isScalable" .) "true" -}}
 {{- if $isDistributed -}}
 {{- $schedulerAddress = printf "%s.%s.svc.%s:%s" (include "loki.querySchedulerFullname" .) .Release.Namespace .Values.global.clusterDomain (.Values.loki.server.grpc_listen_port | toString) -}}
+{{- else if $isSimpleScalable -}}
+{{- $schedulerAddress = printf "query-scheduler-discovery.%s.svc.%s.:%s" .Release.Namespace .Values.global.clusterDomain (.Values.loki.server.grpc_listen_port | toString) -}}
```
Comment (DylanGuedes) [inline production/helm/loki/templates/_helpers.tpl:950]: shouldn't it use a `read` service instead?

Comment (maksym-iv) [reply production/helm/loki/templates/_helpers.tpl:950]: I believe it should be the address of the backend, since in SSD mode it is 
Additionally, before Loki `3.0.0` release it was set to `query-scheduler-discovery` - [ref](https://github.com/grafana/loki/blob/v2.9.7/production/helm/loki/templates/_helpers.tpl#L804-L806)

But there was some more clarifications in the [issue](https://github.com/grafana/loki/issues/12506#issuecomment-2050700539), I'll do a bit more testing today and see if that could be left out empty. I had errors in the reader and writer when upgrading, but will check if I can resolve it.

Comment (DylanGuedes) [reply production/helm/loki/templates/_helpers.tpl:950]: oops, yes, the `backend` target :smile:

**Comment by maksym-iv** [comment]
Closing since the original appears to be a valid behaviour.
Described the original issue and how I solved it in the issue comment https://github.com/grafana/loki/issues/12506#issuecomment-2051430936
